cg/meta/archive/models.py

```diff
@@ -0,0 +1,30 @@
+"""Contains base models to be inherited from in other archive software files."""
+from typing import List
+
+from housekeeper.store.models import File
+from pydantic.v1 import BaseModel
+
+from cg.store.models import Sample
+
+
+class FileTransferData(BaseModel):
+    """Base class for classes representing files to be archived."""
+
+    @classmethod
+    def from_models(cls, file: File, sample: Sample):
+        """Instantiates the class from a File and Sample object."""
+        raise NotImplementedError
+
+
+class ArchiveHandler:
+    """Base class for classes handling different archiving programs."""
```

Comment:
Could potentially be called `ArchiveHandlerInterface` if we want to make it super clear that is an abstract base class (the closest thing Python has to interfaces)

Reply:
Maybe just `ArchiveInterface`?